app/models/primer.rb

```diff
@@ -1,9 +1,13 @@
 class Primer < ActiveRecord::Base
 
   has_many :marker_assays_A, class_name: 'MarkerAssay',
-             foreign_key: 'primer_a'
+             foreign_key: 'primer_a_id'
   has_many :marker_assays_B, class_name: 'MarkerAssay',
-           foreign_key: 'primer_b'
+             foreign_key: 'primer_b_id'
+
+  def marker_assays
+    marker_assays_A | marker_assays_B
```

Comment:
Does this method run two queries and join the results ruby-side or does it combine them in the DB. Asking out of pure curiosity.